optimum/habana/transformers/generation/utils.py

```diff
@@ -1079,6 +1080,8 @@ def greedy_search(
         hb_profer = HabanaProfile(warmup=profiling_warmup_steps, active=profiling_steps)
         hb_profer.start()
         this_peer_finished = False  # used by synced_gpus only
+
+        enable_trim_logits = "trim_logits" in set(inspect.signature(self.forward).parameters.keys())
```

Comment:
Please check my PR https://github.com/huggingface/optimum-habana/pull/396. I returned the new model kwargs in prepare input for generation function inside modeling_llama.py and avoided the if else conditions in utils.py. Also since this is model specific change, we should do there other wise I think it can break other models if this argument is set to True for other models.

Reply:
Fully agree with @puneeshkhanna

Reply:
Thanks for advice, changed accordingly.